experiments/experiment_utils.py

```diff
@@ -398,3 +386,83 @@ def get_datasets(dataset_name, spark: SparkSession, partition_num: int):
 
     return train, test, user_features
 
+
+def get_spark_configs_as_dict(spark_conf: SparkConf):
+    return {
+        "spark.driver.cores": spark_conf.get("spark.driver.cores"),
+        "spark.driver.memory": spark_conf.get("spark.driver.memory"),
+        "spark.memory.fraction": spark_conf.get("spark.memory.fraction"),
+        "spark.executor.cores": spark_conf.get("spark.executor.cores"),
+        "spark.executor.memory": spark_conf.get("spark.executor.memory"),
+        "spark.executor.instances": spark_conf.get("spark.executor.instances"),
+        "spark.sql.shuffle.partitions": spark_conf.get(
+            "spark.sql.shuffle.partitions"
+        ),
+        "spark.default.parallelism": spark_conf.get(
+            "spark.default.parallelism"
+        ),
+    }
+
+
+def check_number_of_allocated_executors(spark: SparkSession):
```

Comment:
As I remember, this function may work incorrectly on some clusters due to redrecting of WebUI to proxy of a cluster manager. I think it should be disabled by default and can be enabled through setting an env variable.